lesson17/index.js

```diff
@@ -0,0 +1,171 @@
+const url = "https://myjson.dit.upm.es/api/bins/8b9d";
+const main = document.querySelector("main");
+
+const controlSlide = (btnType) => {
+  const currentSlide = document.querySelector('[data-hidden="false"]');
+  if(btnType === "prev") {
+    // 今のcurrentのスライドの一つ前の要素をcurrentにする
+    const prevSlide = currentSlide.previousElementSibling;
+    currentSlide.setAttribute("data-hidden","true");
+    prevSlide.setAttribute("data-hidden","false");
+    controlBtnBehavior();
+  } else {
+    // 今のcurrentのスライドの一つ次の要素をcurrentにする
+    const nextSlide = currentSlide.nextElementSibling;
+    currentSlide.setAttribute("data-hidden","true");
+    nextSlide.setAttribute("data-hidden","false");
+    controlBtnBehavior();
+  }
+  updateFractionNum();
+}
+
+const slideImgTemplate = (index, slide) => {
+  const div = document.createElement("div");
+  div.classList.add("sliderImg");
+  div.id = `slide-${index}`;
+  if(index === 0){
+    div.setAttribute("data-hidden", "false");
+  } else {
+    div.setAttribute("data-hidden", "true");
+  }
+  const img = document.createElement("img");
+  img.src = slide.img;
+  div.appendChild(img);
+  return div;
+};
+
+const initSlides = (slides)=> {
+  const slider = document.createElement("div");
+  slider.classList.add("slider");
+  slider.id = "js-slider"
+  const sliderWrapper = document.createElement("div");
+  sliderWrapper.classList.add("sliderWrapper");
+  sliderWrapper.id = "js-sliderWrapper"
+  slider.appendChild(sliderWrapper);
+  main.appendChild(slider);
+
+  const fragment = document.createDocumentFragment();
+  for (const [index, slide] of slides.entries()) {
+    fragment.appendChild(slideImgTemplate(index, slide));
+    sliderWrapper.appendChild(fragment);
+  }
+  createSliderBtns();
+}
+
+const setBtnEvent = () => {
+  const prevBtn = document.getElementById("js-slider-prevBtn");
+  const nextBtn = document.getElementById("js-slider-nextBtn");
+  prevBtn.addEventListener("click", ()=> {
+    controlSlide('prev');
+  })
+  nextBtn.addEventListener("click", ()=> {
+    controlSlide('next');
+  })
+}
+
+const createSliderBtns = () => {
+  const prevBtn = document.createElement("div");
+  prevBtn.classList.add("sliderBtn","prev", "is-disabled");
+  prevBtn.id = "js-slider-prevBtn";
+  const nextBtn = document.createElement("div");
+  nextBtn.classList.add("sliderBtn","next");
+  nextBtn.id = "js-slider-nextBtn";
+  main.querySelector(".slider").appendChild(prevBtn);
+  main.querySelector(".slider").appendChild(nextBtn);
+  setBtnEvent();
+}
```

Comment:
IMO
You may need to modify the css, but you can try this idea.
It's just an image, so I hope it conveys the atmosphere.
```suggestion
const createSliderBtns = () => {
  const sliderDirections = ["previous", "next"];
  sliderDirections.forEach((sliderDirection) => {
    const button = document.createElement("div");
    button.classList.add("sliderBtn", `${sliderDirection}`);
    button.id = `js-slider-${sliderDirection}Btn`;
    button.value = `${sliderDirection}`;
    sliderDirection === "previous" && button.classList.add("is-disabled");
    main.querySelector(".slider").appendChild(button);
  });
  setBtnEvent();
};
```

Reply:
[57c3b34](https://github.com/hippopo-moo/js-handson-work/pull/19/commits/57c3b3495ce1ba7ed897cf70bbf41d058f42fdf1)

Thank you for your comment!
I used Monaka-san's code as a reference and made some modifications!